pkg/block/mem/adapter.go

```diff
@@ -76,7 +76,7 @@ func getKey(obj block.ObjectPointer) string {
 	if obj.IdentifierType == block.IdentifierTypeFull {
 		return obj.Identifier
 	}
-	return fmt.Sprintf("%s:%s", obj.StorageNamespace, obj.Identifier)
+	return fmt.Sprintf("%s:%s", strings.TrimRight(obj.StorageNamespace, "/"), obj.Identifier)
```

Comment:
there was an issue with the cache key?

Reply:
Yap, we would sometime get here with a trailing slash and sometimes don't. It points to the same object, so I made this alignment.